apps/web-giddh/src/app/userDetails/components/subscription/subscription.component.ts

```diff
@@ -0,0 +1,399 @@
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SubscriptionComponent implements OnInit, OnDestroy {
+    /** This will hold local JSON data */
+    @Input() public localeData: any = {};
+    /** This will hold common JSON data */
+    @Input() public commonLocaleData: any = {};
+    /**  This will use for companies list expansion in accordian */
+    @ViewChild(MatAccordion) accordion: MatAccordion;
+    /** This will use for move company in to another company  */
+    @ViewChild("moveCompany", { static: false }) public moveCompany: any;
+    /** This will change the search bar width dynamically */
+    public menuOneWidth: Number = 4;
+    public menuTwoWidth: Number = 12;
+    /** This will change the rowspan of main content and plan list dynamically */
+    public sideBarBoxLength: Number = 15;
+    public sideBarBoxWidth: Number = 4;
+    public mainContentWidth: Number = 12;
+    /** This will change the height of plan list dynamically */
+    public rowLength: Number = 180;
+    /** This will change the length of overall summary box dynamically */
+    public overallSummaryTopRow: Number = 6;
+    public overallSummaryBottomRow: Number = 4;
+    /* This will hold list of subscriptions */
+    public subscriptions: SubscriptionsUser[] = [];
+    /** Observable to listen for subscriptions */
+    public subscriptions$: Observable<SubscriptionsUser[]>;
+    /* This will hold the Subscription to use in selected subscriptions */
+    public selectedSubscription: any;
+    /** This will hold the plan list */
+    public plansList: any[] = [];
+    /* This will hold the companies to use in selected company */
+    public selectedCompany: any;
+    /* This will hold if plan option is choosen */
+    public isPlanShow: boolean = false;
+    /** Stores the current searched subscription */
+    public searchSubscription: FormControl = new FormControl();
+    /** To destroy observers */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** True if api call in progress */
+    public showLoader: boolean = false;
+    /** Image path variable */
+    public imgPath: string = '';
+    /** This will filter the dropdowns of expiry and transaction balance in subscription list */
+    public filters: any = { plan: '', expiration: '', transactionBalance: '' };
+    /** This will stores the seleceted user plans */
+    public seletedUserPlans: SubscriptionsUser;
+    /** This will stores the subscription plans  */
+    public subscriptionPlan: CreateCompanyUsersPlan;
+    /** This will send the subscription object for buy and renew plan */
+    public subscriptionRequestObj: SubscriptionRequest = {
+        planUniqueName: '',
+        subscriptionId: '',
+        userUniqueName: '',
+        licenceKey: ''
+    };
+    /** This will hold the login user */
+    public loggedInUser: UserDetails;
+    /** This will store the static data in expiry list */
+    public expiringList: any[] = [
```

Comment:
this should be based on active language, i will do this, let me know once all other changes are done